src/navigators/createMaterialBottomTabNavigator.js

```diff
@@ -18,19 +18,33 @@ class BottomNavigationView extends React.Component<Props> {
     return options.tabBarColor;
   };
 
+  _isVisible() {
+    const {
+      navigation,
+      descriptors,
+    } = this.props;
+    const {state} = navigation;
+    const route = state.routes[state.index];
+    const options = descriptors[route.key].options;
+    return options.tabBarVisible == null || options.tabBarVisible;
+  }
+
   render() {
     const {
       activeTintColor,
       inactiveTintColor,
       navigation,
       // eslint-disable-next-line no-unused-vars
       descriptors,
+      barStyle
       ...rest
     } = this.props;
-
+    
+    const extraStyle = { display: this._isVisible() ? null : 'none' };
```

Comment:
This won't be enough. We'd also want to check if the `tabBarVisible` option is a boolean to avoid always overriding `barStyle`. With the current code, `barStyle: { display: 'none' }` will stop working. I'll post the code here again:

```js
const isVisible = this._isVisible();
const extraStyle = typeof isVisible === 'boolean' ? { display: isVisible ? 'auto' : 'none' } : null;
```

We'd also need to replace `return options.tabBarVisible == null || options.tabBarVisible;` with `return options.tabBarVisible;` above

Reply:
I see what you mean now. Thanks.